### DISCUSSION TITLE
Is it okay to sync wallet automatically after a given time frame?

### DISCUSSION THREAD

**Discussion by ConorOkus** [bitcoindevkit/bdk#947] [Q&A]
Repost from Discord by user `Roy0Anonymous`

**Comment by ConorOkus** [comment]
**Francis Pouliot:**

"We do resyncs based on a combimation of time, or actions in the wallet (navigating to transactions view, navigating to homepage which has the balance, navigating to address list, swipe down action while on homepage). I think for time we were aimimg for every 10 seconds the user stays on the homepage. Not sure, will check.

Another idea we have is to use websocket from a mempool.space server to receiving real-time notifications of payments to specific addresses, which we cam display immediately and trigger a resync when we receive a notification. Not implemented but considering it, this is mostly useful for in-person payments where the recipient wants the "gratification" of seeing a new transaction come in instantly without having to refresh"

**Comment by thunderbiscuit** [comment]
This is an interesting question and I want to add even more food for thoughts on it.

An aspect of sync we often forget is the privacy implications, and developers of end-user applications should consider their whole stack and make sure they understand the effects of a sync and if/where/to whom this leaks data. Different "syncs" have different tradeoffs. Some applications even explain those to their users, elevating the knowledge level for everyone involved. Here are a few of them:

1. Syncs to Electrum servers imply giving away addresses to those servers. Using a public server is therefore quite different than using a private one. There is a whole range of trust there as well (some servers might not be directly controlled by the user but still considered "trusted" by them, etc.) Related to this is whether this Electrum server can associate these addresses with an IP address; using Tor mitigates this issue. In short, public vs private Electrum servers tradeoffs work across two dimensions: revealing  bitcoin addresses a user is interested in, and revealing the IP addresses interested in these bitcoin addresses. 
2. Public Electrum servers can potentially track the timing of syncs from specific IP addresses, and of course the size of the transactions pertaining to the addresses in question. Mitigation here can be accomplished by using a pool of Electrum servers and querying them randomly (or of course by using a trusted/private server).
3. Syncs using compact block filters have a very different privacy profile, potentially helping privacy issues while requiring more bandwidth and size of download.
